CreateSnapshot/src/main/java/org/opensearch/migrations/CreateSnapshot.java

```diff
@@ -135,7 +135,8 @@ public static void main(String[] args) throws Exception {
     private ICreateSnapshotContext context;
 
     public void run() {
-        var client = new OpenSearchClient(arguments.sourceArgs.toConnectionContext());
+        var clientFactory = new OpenSearchClientFactory(null);
```

Comment:
Passing in `null` here seems a bit wonky.  See my comment about an alternative interface, curious what you think.